src/components/CharacterItem/CharacterItem.test.tsx

```diff
@@ -24,30 +25,43 @@ const withoutBmiCharacter: ModifiedCharacter = {
 
 describe('CharacterItem', () => {
   test('renders a character item with bmi if they have', () => {
-    render(<CharacterItem character={withBmiCharacter} />)
+    render(
+      <UnorderedList>
+        <CharacterItem character={withBmiCharacter} />
+      </UnorderedList>,
+    )
     const character = screen.getByText(withBmiCharacter.name)
     expect(character).toBeInTheDocument()
     const bmi = screen.getByText(/bmi/i)
     expect(bmi).toBeInTheDocument()
   })
 
   test('renders a character item witout bmi if not', () => {
-    render(<CharacterItem character={withoutBmiCharacter} />)
+    render(
+      <UnorderedList>
+        <CharacterItem character={withoutBmiCharacter} />
+      </UnorderedList>,
+    )
     const bmi = screen.queryByText(/bmi/i)
     expect(bmi).not.toBeInTheDocument()
   })
 
   test('show details when users click the button and hide them when users click it again', async () => {
-    render(<CharacterItem character={withBmiCharacter} />)
-    const viewDetailsButton = screen.getByRole('button')
+    render(
+      <UnorderedList>
+        <CharacterItem character={withBmiCharacter} />
+      </UnorderedList>,
+    )
+    const viewDetailsButton = screen.getByRole('button', { name: /view details/i })
 
     // show details
     userEvent.click(viewDetailsButton)
     const height = await screen.findByText(/height/i)
     expect(height).toBeInTheDocument()
+    const hideDetailsButton = await screen.getByRole('button', { name: /hide details/i })
 
     // hide details
-    userEvent.click(viewDetailsButton)
+    userEvent.click(hideDetailsButton)
     const noHeight = await screen.queryByText(/height/i)
     expect(noHeight).not.toBeInTheDocument()
```

Comment:
>I'm afraid I'm not following https://github.com/Lada496/star-wars-list/issues/15 : Does it mean that I only need to >check there are not height element after clicking 'view details' button?

Your implementation was no problem but I wanted to point out that you could remove the duplicate like the following. The end result is pretty much the same so it's not a big deal tho

```typescript
   // assert - grab all expected elements to be used in the test
    const viewDetailsButton = screen.getByRole('button', { name: /view details/i })
    const height = await screen.queryByText(/height/i)

   // act - interact by pressing button
    userEvent.click(viewDetailsButton)

    // assert - make sure the height element is in the screen
    expect(height).toBeInTheDocument()


    // assign - hide details
    const hideDetailsButton = await screen.getByRole('button', { name: /hide details/i })

    // act
    userEvent.click(hideDetailsButton)

    // assert - here, you can use the same "height" variable to perform the expect()
    expect(height).not.toBeInTheDocument()

```